### PR TITLE
Feature time of day condition #6465

### PR DESCRIPTION
**Pull request by Eseraz**
As suggested in issue #6465, this adds a new type of condition that checks if the current time is in a certain time span. The time span can be configured by the user through a start time and an end time. The condition is able to wrap around the midnight mark, so you can specify times from 19.00 to 01.00 for example.

In Paper UI the selection of conditions looks like this:
condition selection
And if you click on 'it is a certain time of day' you will get the configuration dialogue for the newly added TimeOfDayCondition.
configuration of time condition

### CODE REVIEW

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    private static final String END_TIME = "endTime";
+
+    public TimeOfDayConditionHandler(Condition condition) {
+        super(condition);
+    }
+
+    @Override
+    public boolean isSatisfied(Map<String, Object> inputs) {
+
+        String startTimeConfig = (String) module.getConfiguration().get(START_TIME);
```
Comment (davidgraeff) [inline bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: An automation handler is recreated on configuration change. Please extract configuration in the contructor.

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        String endTimeConfig = (String) module.getConfiguration().get(END_TIME);
+        if (startTimeConfig == null || endTimeConfig == null) {
+            logger.error("Time condition with id {} is not well configured: startTime={}  endTime = {}", module.getId(),
```
Comment (davidgraeff) [inline bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: The *error* log level is used for serious events only. A user configuration mistake is maximum a warning.

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+                    startTimeConfig, endTimeConfig);
+            return false;
+        }
+
+        LocalTime currentTime = LocalTime.now().truncatedTo(ChronoUnit.MINUTES);
+        LocalTime startTime = LocalTime.parse(startTimeConfig).truncatedTo(ChronoUnit.MINUTES);
```
Comment (davidgraeff) [inline bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: Do this conversion in the constructor as well, e.g. can be assigned to final fields.

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java (hunk 1 above)
Comment (Eseraz) [reply bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: Good to know! It is now :)

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java (hunk 2 above)
Comment (Eseraz) [reply bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: Ok!

File: bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java (hunk 3 above)
Comment (Eseraz) [reply bundles/automation/org.eclipse.smarthome.automation.module.timer/src/main/java/org/eclipse/smarthome/automation/module/timer/handler/TimeOfDayConditionHandler.java]: Done!

**Pull request merged**